markdown/blog/shai-hulud-postmortem.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Shai-Hulud — What happened, how we fixed it, and what we learned
+date: 2025-11-24T16:56:08.828Z
+type: Communication
+tags:
+  - Security
+  - Incident
+  - Postmortem
+cover: /img/posts/shai-hulud-postmortem/shai-hulud-banner.webp
+authors:
+  - name: Ashish Padhy
+    photo: /img/avatars/shurtugal.webp
+    link: https://www.linkedin.com/in/ashish-padhy3023
+    byline: AsyncAPI Maintainer
+excerpt: 'A postmortem of the supply-chain compromise that affected published packages on npm and related registries.'
+---
+
+On November 24, 2025, our systems experienced a significant security incident that impacted our dependents and users. This postmortem aims to provide a detailed account of what happened, how we addressed the issue, and the lessons we learned to prevent future occurrences.
+
+## Incident Timeline
+
+ **Mon, 24 Nov 2025**
+  - `03:58:47 UTC` — Pushing of malicious package versions to npm registry.
+
+  - `10:02:00 UTC` — [Report by Charlie Eriksen](https://www.aikido.dev/blog/shai-hulud-strikes-again-hitting-zapier-ensdomains) from Aikido about compromised packages.
+  - `10:32:00 UTC` — Acknowledgment of the issue by AsyncAPI team, and communication initiated to inform the community.
+  - `10:42:00 UTC` — Initial investigation and revoking of NPM tokens.
+  - `10:58:00 UTC` — Reported the incident to npm security team and started deprecating affected packages as unpublishing is not allowed with dependent packages.
+  - `11:46:00 UTC` — Action taken by NPM security team to unpublish the malicious packages.
+  - `13:33:00 UTC` — Suspected leak found in [vs-asyncapi-preview](https://github.com/asyncapi/vs-asyncapi-preview), as malicious version (1.0.1) was pushed to [OpenVSX Registry](https://open-vsx.org/extension/asyncapi/asyncapi-preview) through report from [Yusuke Sugamiya](https://x.com/DNPP)
+  - `13:45:00 UTC` — Revoked OpenVSX token and reported to OpenVSX security team.
+  - `14:30:00 UTC` — OpenVSX team unpublished the malicious package version.
```

Comment:
I got email from them exactly at 13:52 UTC that version was unpublished

Reply:
Done.